js/views/pageNavVw.js

```diff
@@ -368,16 +371,171 @@ module.exports = Backbone.View.extend({
+        var secondaryColor =   parseInt($($("label[for='"+themeId+"']")[0]).data('secondary-color').replace("#","0x"));
+        var backgroundColor =  parseInt($($("label[for='"+themeId+"']")[0]).data('background-color').replace("#","0x"));
+        var textColor =   parseInt($($("label[for='"+themeId+"']")[0]).data('text-color').replace("#","0x"));
+        var header = $($("label[for='"+themeId+"']")[0]).data('header');
+
+
+        self.model.set('primary_color', primaryColor);
+        self.model.set('secondary_color', secondaryColor);
+        self.model.set('text_color', backgroundColor);
+        self.model.set('background_color', textColor);
+        //TODO upload Image header
+        //From profile api : header= the hash of the header image. must have been previously uploaded using the upload_image api call. (40 character hex string)
+        //self.model.set('header', header);
+
+    }
+
+    $.each(this.model.attributes,
+            function(i,el) {
+                if(i == "country") {
+                    profileFormData.append("location",el);
+                }
+                if(i == "name" || i == "handle" || i =="about"|| (themeSelected && (i == "primary_color" || i == "secondary_color" || i == "text_color"|| i =="background_color" ))) {
+                    profileFormData.append(i,""+el);
+                } else if(i == "tempAvatar") {
+                    var imageURI = $('#avatarUploadDiv').cropit('export', {
+                        type: 'image/jpeg',
+                        quality: 0.75,
+                        originalSize: false
+                    });
+                    if(imageURI) {
+                        imageURI = imageURI.replace(/^data:image\/(png|jpeg);base64,/, "");
+                        uploadImageFormData.append('image', imageURI);
+                    }
+                } else {
+                    settingsFormData.append(i,el);
+                }
+
+            }
+        );
+
+      settingsFormData.append("shipping_addresses","");
+
+
+   var submit = function(img_hash) {
+            if(img_hash) {
+                profileFormData.append("avatar",img_hash);
+            }
+
+            $.ajax({
+                type: "POST",
+                url: server + "settings",
+                contentType: false,
+                processData: false,
+                data: settingsFormData,
+                success: function(data) {
+                    if(JSON.parse(data).success) {
+                        $.ajax({
+                            type: "POST",
+                            url: server + "profile",
+                            contentType: false,
+                            processData: false,
+                            data: profileFormData,
+                            success: function(data) {
+                                 if(data.success === true) {
+                                     if(img_hash) {
```

Comment:
Ignore the comment above, I see you're making this check in the upload image function.